charts/nginx-gateway-fabric/templates/deployment.yaml

```diff
@@ -15,14 +15,17 @@ spec:
     metadata:
       labels:
       {{- include "nginx-gateway.selectorLabels" . | nindent 8 }}
-      {{- if .Values.metrics.enable }}
```

Comment:
If metrics is not enabled and there are no podAnnotations, we end up with:

```
  template:
    metadata:
      labels:
        app.kubernetes.io/name: nginx-gateway-fabric
        app.kubernetes.io/instance: helm-dev
      annotations:
```

this is technically fine, but it does look a little weird.

@sjberman do you think it's worth it to add another condition here to check if metrics are enabled or podAnnotations are present to prevent an empty annotations section? I'm on the fence

Reply:
I think I'm okay with it being empty, not a huge deal to me.

Reply:
I think it makes sense to add
```
{{- if or .Values.nginxGateway.podAnnotations .Values.metrics.enable }}
```
before `annotations` 😬

Reply:
Thanks for the suggestion @lucacome , this has been added to the PR.